usagov/usagov-2021
# Usagov 646 move elected officials

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-646

## Description


Appended to hook usagov_theme_suggestions_alter found in usagov.theme to allow us to create twig templates for code that was in the body field in the cms. Essentially this hook gets the current pages alias and appends it to the string 'field__node__body__basic_page_'. The new string is then added as a twig template suggestion. 

Item(s) of note: 
- For some reason the string in the theme file has to use underscores NOT dashes. The actual twig template can use dashes but it took hours of debugging to find that Drupal will only find the template if the hook uses underscores. 
- A previous implementation of this (see commit history or [Branch 644)](https://github.com/usagov/usagov-2021/tree/USAGOV-644-Move-States-From-CMS) used an include statement in the field--node--basic_page. Though this allowed the English and Spanish templates to be in the same file (which this implementation does not) it lead to a cluttered basic_page template. 
- As mentioned above, English and Spanish twig templates have to be separate files. The hook works by taking the current path alias. Because of this the English and Spanish pages have different hooks. 
- For the hook to be called the body in the cms cannot be empty. Any information in the body will be overwritten by the twig file but the hook is only ever called if there is data to begin with in the body field in the cms. 

### To Test
1. You may need to rebuild your cache/ theme registry since there are two new files. 
2. Once you do navigate to /elected-officials. 
3. If you do not have twig debugging set up:

- Navigate to web/themes/custom/usagov/templates/field--node--body--basic-page-elected-officials.html.twig
- Add a print statement of your choosing
- Validate you can see the print statement and that the body is being 

3. If you do have twig debugging set up:
- Inspect the body element 
- Verify the element is being populated from the template (i.e you should see:  )  

## Checklist for the Developer



## Checklist for the Peer Reviewers
